### PR TITLE
Move CSV to 0.4.0

### LINKED ISSUE #151: Correct mention of Ingress Controllers on CSV
We mention we deploy an Ingress Controller in our CSV:

```
╰─ grep -i "ingress controller" deploy/olm-catalog/nexus-operator/0.3.0/nexus-operator.v0.3.0.clusterserviceversion.yaml 
    * Creates an [Ingress controller](https://kubernetes.io/docs/concepts/services-networking/ingress/) in Kubernetes (1.14+) environments to expose the application to the world
```

However we really only deploy ingresses, no ingress controllers. This needs to be changed on the next release's CSV.

### PR DESCRIPTION
**Pull request by LCaparelli**
Fix #151

Signed-off-by: Lucas Caparelli <lucas.caparelli112@gmail.com>

### CODE REVIEW

**Comment by LCaparelli** [comment]
Oops, something's wrong. Looking

**Comment by LCaparelli** [comment]
Not sure what's going on, the e2e tests ran fine locally. At the CI it just times out waiting for the first test's deployment. Not to mention I didn't even change anything related to the e2e tests, so this seems pretty weird. Any ideas?

**Comment by Kaitou786** [comment]
Should the image tag in `deploy/operator.yaml` be changed to 0.4?

**Comment by ricardozanini** [comment]
> Should the image tag in `deploy/operator.yaml` be changed to 0.4?

yes it should

File: deploy/olm-catalog/nexus-operator/0.4.0/nexus-operator.v0.4.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+              }
+            },
+            "useRedHatImage": false
+          }
+        }
+      ]
+    capabilities: Seamless Upgrades
+    categories: Developer Tools
+    certified: "false"
+    containerImage: quay.io/m88i/nexus-operator:0.3.0
```
Comment (ricardozanini) [inline deploy/olm-catalog/nexus-operator/0.4.0/nexus-operator.v0.4.0.clusterserviceversion.yaml:38]:
```suggestion
    containerImage: quay.io/m88i/nexus-operator:0.4.0
```

File: deploy/olm-catalog/nexus-operator/0.4.0/nexus-operator.v0.4.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.annotations['olm.targetNamespaces']
+                - name: POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: nexus-operator
+                image: quay.io/m88i/nexus-operator:0.3.0
```
Comment (ricardozanini) [inline deploy/olm-catalog/nexus-operator/0.4.0/nexus-operator.v0.4.0.clusterserviceversion.yaml:180]:
```suggestion
                image: quay.io/m88i/nexus-operator:0.4.0
```

File: deploy/olm-catalog/nexus-operator/manifests/nexus-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -41,7 +41,7 @@ metadata:
     repository: https://github.com/m88i/nexus-operator
     support: m88i Labs
     tectonic-visibility: ocs
-  name: nexus-operator.v0.3.0
+  name: nexus-operator.v0.4.0
```
Comment (ricardozanini) [inline deploy/olm-catalog/nexus-operator/manifests/nexus-operator.clusterserviceversion.yaml:44]: The image name should be changed as well :P

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Just update the image version and we should be ready to go.

**Comment by LCaparelli** [comment]
> Should the image tag in deploy/operator.yaml be changed to 0.4?

>> yes it should

If we change it users will need to build the 0.4 image locally since there's no 0.4 on Quay, no? I think we should only change it on the next release, especially considering `deploy/operator.yaml` will be used to install the operator remotely without cloning the repo in #154

But I wonder if this has to do with the failure of the e2e tests, since on previous releases `deploy/operator.yaml` was indeed pointed to the next release, not the current one IIRC. If this is the case, we might want to consider having an `operator.yaml` file as one of the assets per release, this way we freeze it per release with a readable URL (as opposed to one pointing to a specific commit hash).

But given we change PullPolicy on our e2e script to `Never` before running, it still would just use whatever image it just built, regardless of its tag, no?

```
 37     echo "---> Updating deployment file to imagePullPolicy: Never"
 38     sed -i.bak 's/imagePullPolicy:\s*Always/imagePullPolicy: Never/g' ./deploy/operator.yaml
```

I even tried building with Alpine:latest locally as base image, but no luck.

> Just update the image version and we should be ready to go.

Yeah, I completely missed these last night, my bad.

**Comment by Kaitou786** [comment]
> If we change it users will need to build the 0.4 image locally since there's no 0.4 on Quay, no? I think we should only change it on the next release, especially considering deploy/operator.yaml will be used to install the operator remotely without cloning the repo in #154

I think we can keep it with the next release version, 0.4 . For #154  we can have a similar procedure we've with the `make install`, i.e. checkout the repository at the latest tag(0.3), and then apply the `operator.yaml`

> But given we change PullPolicy on our e2e script to Never before running, it still would just use whatever image it just built, regardless of its tag, no?

Could be the image is build with 0.4 tag which the deployment is not able to find, hence the deployment not running?

**Comment by LCaparelli** [comment]
> I think we can keep it with the next release version, 0.4 . For #154 we can have a similar procedure we've with the make install, i.e. checkout the repository at the latest tag(0.3), and then apply the operator.yaml

The idea there is to have users run `kubectl apply -f https://bla-bla/operator.yaml` in order to install everything. We first considered a similar remote installation script which would look a lot like what `make install` does, but ultimately just running `kubectl apply <...>`is just simpler and I think we should keep it that way. 

> Could be the image is build with 0.4 tag which the deployment is not able to find, hence the deployment not running?

Ah yes, that makes sense, since the tests are performed by applying the yaml files `deploy` (which points to 0.3, not 0.4). It's probably working on mine because I have 0.3 cached as I didn't rebuild my minikube VM after some tests I was performing with 0.3. Nice one @Kaitou786, thanks! :-D

@ricardozanini @Kaitou786 I'll change `deploy/operator.yaml` for now, but let's continue this discussion on #154 or telegram regarding how we should allow remote installation.

**Comment by Kaitou786** [comment]
Ah, yes @LCaparelli I was thinking if we are going to maintain a separate file with everything in it, the image tag there can be for the last released version of our operator. Just a though :p. We can discuss this more :)

**Comment by LCaparelli** [comment]
> Ah, yes @LCaparelli I was thinking if we are going to maintain a separate file with everything in it, the image tag there can be for the last released version of our operator.

Absolutely, that was one of the approaches I was thinking about too. We just have to keep it off the `deploy` directory due to this (I don't think that's a bad thing, just pointing out the pitfall I just fell in hehe).

**Comment by LCaparelli** [comment]
@ricardozanini @Kaitou786 CI passing now :-)

**Review by Kaitou786 — APPROVED** [review]
Great!

**Comment by ricardozanini** [comment]
@LCaparelli the idea is to not have a separate operator.yaml file since it's alone do nothing. We need RBAC as well.. so the idea is to combine everything in a single file like `nexus-operator.yaml` and make it part of the release.

So users would simply `kubectl apply -f <github_release_url>/0.4.0/nexus-operator.yaml` to install it.

**Review by ricardozanini — APPROVED** [review]

**Comment by LCaparelli** [comment]
> @LCaparelli the idea is to not have a separate operator.yaml file since it's alone do nothing. We need RBAC as well.. so the idea is to combine everything in a single file like `nexus-operator.yaml` and make it part of the release.
> 
> So users would simply `kubectl apply -f <github_release_url>/0.4.0/nexus-operator.yaml` to install it.

@ricardozanini Yes, I understand that. I mention that on the issue hehe :-)

**Pull request merged**